lmnet/tests/executor_tests/test_build_tfds.py

```diff
@@ -0,0 +1,194 @@
+        assert images.shape[1] == config.IMAGE_SIZE[0]
+        assert images.shape[2] == config.IMAGE_SIZE[1]
+        assert images.shape[3] == 3
+
+        assert isinstance(labels, np.ndarray)
+        assert labels.shape[0] == config.BATCH_SIZE
+        assert labels.shape[1] == validation_dataset.num_classes
+
+
+def test_build_tfds_object_detection():
+    environment.setup_test_environment()
+
+    # Build TFDS Dataset
+    config_file = "tests/fixtures/configs/for_build_tfds_object_detection.py"
+    run(config_file, overwrite=True)
+
+    # Check if the builded dataset can be loaded with the same config file
+    expriment_id = "tfds_object_detection"
+    train_run(None, None, config_file, expriment_id, recreate=True)
+
+    # Check if the dataset was build correctly
+    train_data_num = 3
+    validation_data_num = 2
+    config = config_util.load(config_file)
+
+    train_dataset = setup_dataset(TFDSObjectDetection,
+                                  subset="train",
+                                  batch_size=config.BATCH_SIZE,
+                                  pre_processor=config.PRE_PROCESSOR,
+                                  **config.DATASET.TFDS_KWARGS)
+
+    validation_dataset = setup_dataset(TFDSObjectDetection,
+                                       subset="validation",
+                                       batch_size=config.BATCH_SIZE,
+                                       pre_processor=config.PRE_PROCESSOR,
+                                       **config.DATASET.TFDS_KWARGS)
+
+    assert train_dataset.num_per_epoch == train_data_num
+    assert validation_dataset.num_per_epoch == validation_data_num
+
+    assert train_dataset.num_max_boxes == validation_dataset.num_max_boxes
+    num_max_boxes = train_dataset.num_max_boxes
+
+    for _ in range(train_data_num):
+        images, labels = train_dataset.feed()
+        _show_images_with_boxes(images, labels)
+
+        assert isinstance(images, np.ndarray)
+        assert images.shape[0] == config.BATCH_SIZE
+        assert images.shape[1] == config.IMAGE_SIZE[0]
+        assert images.shape[2] == config.IMAGE_SIZE[1]
+        assert images.shape[3] == 3
+
+        assert isinstance(labels, np.ndarray)
+        assert labels.shape[0] == config.BATCH_SIZE
+        assert labels.shape[1] == num_max_boxes
+        assert labels.shape[2] == 5
+
+    for _ in range(validation_data_num):
+        images, labels = validation_dataset.feed()
+        _show_images_with_boxes(images, labels)
+
+        assert isinstance(images, np.ndarray)
+        assert images.shape[0] == config.BATCH_SIZE
+        assert images.shape[1] == config.IMAGE_SIZE[0]
+        assert images.shape[2] == config.IMAGE_SIZE[1]
+        assert images.shape[3] == 3
+
+        assert isinstance(labels, np.ndarray)
+        assert labels.shape[0] == config.BATCH_SIZE
+        assert labels.shape[1] == num_max_boxes
+        assert labels.shape[2] == 5
+
+
+if __name__ == '__main__':
```

Comment:
i am not that familiar with pytest but IIUC this __main__ thing is unnecessary and we should let pytest decide what to run.

Reply:
This was for debugging this script. I've removed it.